src/wrapper_array.jl

```diff
@@ -176,3 +176,43 @@ end
 @propagate_inbounds function Base.setindex!(a::NamedDimsArray, value, inds...)
     return setindex!(parent(a), value, inds...)
 end
+
+############################################
+# show
+
+# 2-arg show, mainly for repr():
+function Base.show(io::IO, A::NamedDimsArray{L,T,N}) where {L,T,N}
+    print(io, "NamedDimsArray(")
+    show(io, parent(A))
+    print(io, ", ", N == 1 ? QuoteNode(L[1]) : L, ")")
```

Comment:
I think for `repr` we probably want the names in the type info, otherwise the names are separated from from the type by the arbitrarily large number of values (whereas in the `showarg`/`summary` case we're not showing the values so don't have this problem).

e.g. even in a simple case
```
julia> nda = NamedDimsArray(rand(2,3), (:x,:y));

julia> repr(nda)
"NamedDimsArray([0.7434604919275618 0.6970999852412016 0.17194407552471547; 0.01565280781692069 0.753384104139615 0.5902236599197668], (:x, :y))"
```
I think it is clearer, in this context, for it to be shown as
```
julia> nda = NamedDimsArray(rand(2,3), (:x,:y));

julia> repr(nda)
"NamedDimsArray{(:x, :y)}([0.7434604919275618 0.6970999852412016 0.17194407552471547; 0.01565280781692069 0.753384104139615 0.5902236599197668])"

Reply:
My thinking was that it's better to present one way of writing things, not two. I agree that `string(array)` for a large array doesn't print something very eyeball-friendly, never has. It does get trimmed in some contexts, e.g. `[NamedDimsArray(rand(Int8, 30),:x) for y in 1:4]`.

Reply:
It's also possible that this should be turned off based on IOContext... here `Int8` is not printed when the container already specifies it:
```
julia> rand(Int8, 3) |> string
"Int8[-121, 12, 63]"

julia> [rand(Int8, 3) for _ in 1:2]
2-element Vector{Vector{Int8}}:
 [19, 127, -37]
 [-110, -124, -52]

julia> Any[rand(Int8, 3) for _ in 1:2]
2-element Vector{Any}:
 Int8[13, -107, 37]
 Int8[85, 45, -111]
```

Reply:
Now turned off based on whether the container has eltype `NamedDimsArray`, which is how Base handles similar things.